cypress/integration/login.js

```diff
@@ -1,26 +1,42 @@
-describe('Login functionality', function() {
+import LoginPage from '../obj/loginPageObj';
+
+const loginPage = new LoginPage();
+
+describe('Login functionality', () => {
     
-    it('Should display validation for empty user after attempted loggin', function () {
-        cy.visit('/')
-        cy.get('.Select.not-valid').should('not.visible')
-        cy.get('[type="submit"]').click()
-        cy.get('.Select.not-valid').should('be.visible')
-    })
+    it('Should display validation for empty user after attempted loggin', () => {
+        loginPage.visit();
+        loginPage.getValidationIndicator().should('not.visible');
+        loginPage.getLoginButton().click();
+        loginPage.getValidationIndicator().should('be.visible');
+    });
 
-    it('Should be able to login with role User', function () {
-        cy.get('[id="loginForm.userId"]').click({force:true})
-        cy.get('[aria-label="Demo User"]').click()
-        cy.get('[id="loginForm.role"]').click({force:true})
-        cy.get('[aria-label="User"]').click()
-        cy.get('[type="submit"]').click()
+    it('Should be able to login with role Team Lead', () => {
+        const roles = [
+            { username: 'Team Lead', length: 2 },
+            { username: 'Manager', length: 5 },
+            { username: 'Accountant', length: 5 },
+            { username: 'Admin', length: 6 }
+        ];
 
-        cy.url().should('include', '/time-logging')
-        cy.get('.page__title').contains('Timesheets')
-        cy.get('.calendar').should('be.visible')
-        cy.get('.tile.form').should('be.visible')
-        cy.get('.user-info__title').contains('Demo User')
-        cy.get('.main-nav').find('li').should('have.length', 1)
-    })
+        roles.map(role => {
+            loginPage.visit();
+            loginPage.getLoginButton().click();
+            loginPage.getLoginFormId('loginForm.userId').click({force:true});
+            loginPage.getUserName('TestCon User 4').click();
+            loginPage.getLoginFormId('loginForm.role').click({force:true});
+            loginPage.getUserName(role.username).click();
+            loginPage.getLoginButton().click();
+
+            cy.url().should('include', '/time-logging');
+            cy.get('.page__title').contains('Timesheets');
+            cy.get('.calendar').should('be.visible');
+            cy.get('.tile.form').should('be.visible');
+            cy.get('.user-info__title').contains('TestCon User 4');
+            cy.get('.main-nav').find('li').should('have.length', role.length);
+            cy.get('[aria-labelledby="timeLoggingId"]').should('have.css', 'color','rgb(64, 76, 237)');
```

Comment:
Goal of T-9 was to refactor all test cases to page object model, this test case still missing Timesheets/timelogging page object refactoring